### PR TITLE
Hash token

### PR DESCRIPTION
**Pull request by KerstenBreuer**
- store Apikey as hash in database
- template for custom errors, usage:
```
from ..errors import ValidationError
...
raise ValidationError("my message")
```
- use ApiKey label and user side id

### CODE REVIEW

File: datameta/security.py [hunk 1]
```diff
@@ -29,12 +29,12 @@
...
 
 def hash_password(pw):
-    """Hash a password and return the salted hash"""
+    """Hash a password/token and return the salted hash"""
     pwhash = bcrypt.hashpw(pw.encode('utf8'), bcrypt.gensalt())
     return pwhash.decode('utf8')
 
 def check_password_by_hash(pw, hashed_pw):
-    """Check a password against a salted hash"""
+    """Check a password/hash against a salted hash"""
```
Comment (lkuchenb) [inline datameta/security.py:37]:
```suggestion
    """Check a password/token against a salted hash"""
```

**Review by MoritzHahn1337 — APPROVED** [review]

File: datameta/errors.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from pyramid.response import Response
+from pyramid.view import view_config
+from pyramid.httpexceptions import HTTPBadRequest
+import json
+
+class ValidationError(Exception):
```
Comment (lkuchenb) [inline datameta/errors.py:28]: I think you can just user a body override of the standard classes here

**Review by lkuchenb — CHANGES_REQUESTED** [review]

File: datameta/security.py (hunk 1 above)
Comment (KerstenBreuer) [reply datameta/security.py:37]: Right

**Comment by KerstenBreuer** [comment]
@lkuchenb can you check that errors.py again?

The new usage is:
```
from ..errors import get_validation_error
...
raise get_validation_error(messages=["my message"])
```

**Review by lkuchenb — APPROVED** [review]

**Review by MoritzHahn1337 — APPROVED** [review]

**Pull request merged**